test/ussd_nurse_rapidpro.test.js

```diff
@@ -439,7 +454,90 @@ describe('app', function() {
+                    it("should loop back without api call", function() {
+                        return tester
+                            .setup.user.addr('27820001001')
+                            .inputs(
+                                {session_event: 'new'}
+                                ,'1' //chooses to start nurse connect
+                                ,'1'  // chooses to sign up
+                                ,'1' //chooses yes to subscribe
+                                ,'0820001004'  // state_msisdn
+                                ,'1' //chooses yes to opt in
+                                ,'12345'  // state_faccode
+                            )
+                            .check.interaction({
+                                state: 'state_faccode',
+                                reply: "Sorry, we don't recognise that code. Please enter the 6- digit facility code again, e.g. 535970:"
+                            })
+                            .run();
+                    });
+                });
+
+                describe("is not on jembi system", function() {
+                    it("should loop back", function() {
+                        return tester
+                            .setup.user.addr('27820001001')
+                            .inputs(
+                                {session_event: 'new'}
+                                ,'1' //chooses to start nurse connect
+                                ,'1'  // chooses to sign up
+                                ,'1' //chooses yes to subscribe
+                                ,'0820001004'  // state_msisdn
+                                ,'1' //chooses yes to opt in
+                                ,'888888'  // state_faccode
+                            )
+                            .check.interaction({
+                                state: 'state_faccode',
+                                reply: "Sorry, we don't recognise that code. Please enter the 6- digit facility code again, e.g. 535970:"
+                            })
+                            .run();
+                    });
+                });
+            });
+
+                describe("unregistered user enters facility code", function(){
+                    it("should go to state_faccname", function() {
+                        return tester
+                            .setup.user.addr('27820001004')
+                            .inputs(
+                                {session_event: 'new'}
+                                ,'1' //chooses to start nurse connect
+                                ,'1'  // chooses to sign up
+                                ,'1' //chooses yes to subscribe
+                                ,'0820001004'  // state_msisdn
+                                ,'1' //chooses yes to opt in
+                                ,'12345' //enters facility code
+                            )
+                            .run();
```

Comment:
We're not checking here if it actually goes to the `state_faccname` state?

Reply:
Thanks for pointing this out, need to add more to make sure it goes to the correct state